scripts/deploy-prometheus.sh

```diff
@@ -0,0 +1,189 @@
+    echo
+    echo "Options:"
+    echo "  -n <namespace>  The namespace used for prometheus service."
+    echo "  -l <label>      The label monitored by prometheus service."
+    echo "  -a <action>     The action to run in this script, deploy or undeploy."
+    echo "  -u <username>   The database username, should have access to the Vertica server metrics."
+    echo "  -p <password>   The database user password."
+    echo "  -d <dbname>     The database name."
+    echo "  -h <usage>      Print help message."
+    exit 1
+}
+
+while getopts "n:l:a:u:p:d:h" opt
+do
+    case $opt in
+        n)
+            NAMESPACE=$OPTARG
+            ;;
+        l)
+            LABEL=$OPTARG
+            ;;
+        a)
+            ACTION=$OPTARG
+            ;;
+        u)
+            USERNAME=$OPTARG
+            ;;
+        p)
+            PASSWORD=$OPTARG
+            ;;
+        d)
+            DBNAME=$OPTARG
+            ;;
+        h) 
+            usage
+            exit 0
+            ;;
+        \?)
+            echo "ERROR: unrecognized option: -$opt"
+            usage
+            exit 0
+            ;;
+    esac
+done
+
+if [ -z "$NAMESPACE" ]
+then
+  NAMESPACE=default
+fi
+
+
+function deploy(){
+  # Create an secret to store prometheus service monitor database username and password.
+  cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: $NAMESPACE
+  name: prometheus-$DBNAME
+data:
+  username: '$(echo -n $USERNAME | base64)'
+  password: '$(echo -n $PASSWORD | base64)'
+type: Opaque
+EOF
+
+  # Create a subscription to the verticadb-operator
+  cat <<EOF | kubectl apply -f -
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: k8s-vertica-prometheus-$DBNAME
+  labels:
+    release: $LABEL
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: $DBNAME
+  namespaceSelector:
+    matchNames:
+      - $NAMESPACE
+  endpoints:
+    - basicAuth:
+        password:
+          key: password
+          name: prometheus-$DBNAME
+        username: 
+          key: username
+          name: prometheus-$DBNAME
+          optional: true
+      interval: 5s
+      path: /v1/metrics
+      port: vertica-http
+      scheme: https
+      tlsConfig:
+        insecureSkipVerify: true
+EOF
+}
+
+function undeploy(){
+  # Delete the service monitor for prometheus service monitor.
+  cat <<EOF | kubectl delete -f -
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: k8s-vertica-prometheus-$DBNAME
+  labels:
+    release: $LABEL
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: $DBNAME
+  namespaceSelector:
+    matchNames:
+      - $NAMESPACE
+  endpoints:
+    - basicAuth:
+        password:
+          key: password
+          name: prometheus-$DBNAME
+        username: 
+          key: username
+          name: prometheus-$DBNAME
+          optional: true
+      interval: 5s
```

Comment:
This (scraping interval) should also be a parameter.

Reply:
addressed in https://github.com/vertica/vertica-kubernetes/pull/1013/commits/4cd17545f32cb85134bfdedc6a51e82a8b5549b1